### DISCUSSION TITLE
Make startRecording() async

### DISCUSSION THREAD

**Discussion by Singe72** [Kirdock/discordjs-voice-recorder#2] [Ideas]
I'm using DisTube for my project and the function to create a connection is asynchronous, so I'm forced to set a timeout because the connection is not done instantly. It would be cool if the `startRecording()` function was asynchronous 😄

**Comment by Kirdock** [comment]
I would not make a function async if it isn't needed.
If you want it async you could basically write
```ts
(async()=> {
  voiceRecorder.startRecording(...)
})();

// or
async function startRecording(connection: VoiceConnection): Promise<void> {
  voiceRecorder.startRecording(connection);
}
```

I'm still not quite sure about your use-case. If the create connection function is async, you can basically `await` it or start the recording inside the `then(()=>...)` of the Proimse.
